happi/cli.py

```diff
@@ -0,0 +1,67 @@
+"""
+This module defines the ``happi`` command line utility
+"""
+# cli.py
+
+import argparse
+import sys
+import happi
+import logging
+import coloredlogs
+
+logger = logging.getLogger(__name__)
+
+# Argument Parser Setup
+parser = argparse.ArgumentParser(description='happi command line tool')
+
+parser.add_argument('--path', type=str,
+                    help='path to happi configuration file')
+parser.add_argument('--search', nargs=argparse.REMAINDER,
+                    help='search criteria: [--search] field value')
+parser.add_argument('--verbose', '-v', action='store_true',
+                    help='Show the degub logging stream')
+parser.add_argument('--version', '-V', action='store_true',
+                    help='Current version and location '
+                    'of Happi installation.')
+
+# TODO: Implement add and edit options
+# parser.add_argument('--add', action='store_true', # needs work
+#                    help='add a device')
+# parser.add_argument('--edit', action='store_true',
+#                    help='edit a device')
+
+
+def happi_cli(args):
+    args = parser.parse_args(args)
+
+    # Logging Level handling
+    if args.verbose:
+        level = "DEBUG"
+        shown_logger = logging.getLogger('happi')
+    else:
+        shown_logger = logging.getLogger()
+        level = "INFO"
+    coloredlogs.install(level=level, logger=shown_logger,
+                        fmt='[%(asctime)s] - %(levelname)s -  %(message)s')
+    logger.debug("Set logging level of %r to %r", shown_logger.name, level)
+
+    # Version endpoint
+    if args.version:
+        print(f'Happi: Version {happi.__version__} from {happi.__file__}')
+        return
+    logger.debug('Command line arguments: %r' % args)
+
+    client = happi.client.Client.from_config(cfg=args.path)
+
+    if args.search:
+        if args.search[0] is 'z':
```

Comment:
Instead of checking just if it is a `z` keyword try `args.search[1].is_number()` this will tell you if the key you have could be converted to a number or not. If it looks like a number it probably is 😉